src/Servers/Kestrel/Core/src/Internal/Http/ChunkWriter.cs

```diff
@@ -44,6 +36,18 @@ public static int BeginChunkBytes(int dataCount, Span<byte> span)
             return count;
         }
 
+        internal static int GetBeginChunkByteCount(int count, out int total, out int shift)
+        {
```

Comment:
I'm not a fan of how I refactored this, but couldn't find a better way. Thoughts?

Reply:
At a minimum, I would keep out param method private and have the internal version of the method without the out param.

Have you verified GetBeginChunkByteCount works for the all ints between 1 and int.MaxValue? Because it only had to work for ints up to 4096 before, right? It might be worth adding unit tests for both this and BeginChunkBytes.

Reply:
No, this had to work for all values up to Int.MaxValue. Tests are here: https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Core/test/ChunkWriterTests.cs#L39

Reply:
I'm more concerned about the logic I added to Http1OutputProducer with regards to all values.